packages/serverpod_serialization/test/json_serialization_test.dart

```diff
@@ -1,28 +1,177 @@
 import 'package:serverpod_serialization/serverpod_serialization.dart';
 import 'package:test/test.dart';
 
-void main() {
-  test('Given json data when encoded to string then output is stringified json',
-      () {
-    var jsonData = {'a': 1, 'b': 2};
+class _User implements SerializableModel, ProtocolSerialization {
+  final String name;
+  final String password;
 
-    var stringifiedJson = SerializationManager.encode(jsonData);
-    expect(stringifiedJson, '{"a":1,"b":2}');
+  _User({
+    required this.name,
+    required this.password,
   });
 
+  @override
+  Map<String, String> toJson() {
+    return {
+      'name': name,
+      'password': password,
+    };
+  }
+
+  @override
+  Map<String, String> toJsonForProtocol() {
+    return {
+      'name': name,
+    };
+  }
+}
+
+void main() {
   test(
-      'Given json data when encoded to formatted string then output is formatted stringified json',
-      () {
-    var jsonData = {'a': 1, 'b': 2};
+    'Given simple JSON data, when encoded using encode method, then output is stringified JSON',
+    () {
+      var jsonData = {'a': 1, 'b': 2};
+      var stringifiedJson = SerializationManager.encode(jsonData);
+      expect(stringifiedJson, '{"a":1,"b":2}');
+    },
+  );
 
-    var stringifiedJson = SerializationManager.encode(
-      jsonData,
-      formatted: true,
-    );
-    expect(stringifiedJson, '''
+  test(
+    'Given simple JSON data, when encoded using encode method with formatting, then output is formatted stringified JSON',
+    () {
+      var jsonData = {'a': 1, 'b': 2};
+      var stringifiedJson = SerializationManager.encode(
+        jsonData,
+        formatted: true,
+      );
+      expect(
+        stringifiedJson,
+        '''
 {
   "a": 1,
   "b": 2
-}''');
+}''',
+      );
+    },
+  );
+
+  group('Given a user object,', () {
+    late _User user;
+
+    setUp(() {
+      user = _User(name: 'John', password: '123');
```

Comment:
Suggestion: assign directly inside of group and remove setUp.

Since nothing changes for the class and we don't modify its state we don't need to re-assign it as part of the `setUp`.